app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt

```diff
@@ -22,6 +24,17 @@ class VoiceCommandsAdapter(val items: ArrayList<String>, val context: Context?)
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
 
         holder?.voiceCommand?.text = items.get(position)
+        holder.voiceCommand.setOnClickListener {
+            val chatMessage = items.get(position)
+            val splits = chatMessage.split("\n".toRegex()).dropLastWhile { it.isEmpty() }
+            val message = splits.joinToString(" ")
+            if (!chatMessage.isEmpty()) {
+                chatPresenter.sendMessage(message, chatMessage)
+                val intent = Intent(context, ChatActivity::class.java)
+                intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+                context?.startActivity(intent)
+            }
+        }
```

Comment:
A better place for this code is inside the ViewHolder class. Inside the view holder class, the data is bound with the view while the view is bound with the holder in the onBindViewHolder().

Reply:
Initially, it was inside ViewHolder class, but I found that it was not working as expected. So, I moved the code inside the BIndViewHolder. I debugged it and found, the onClick was simply not recognizable.

Reply:
You should try debugging and looking into why the code was not working.

Reply:
I tried doing so. Also, I rewrote the code differently, but the result was the same. So, upon google search, I came upon this kind of solution.

Reply:
Google search will definitely guide you, but at the end of the day, you will have to debug it line by line yourself and figure out what's going wrong. I suggest you check the code once again. Check the errors carefully, read about those errors i.e. why and when are they encountered and then find an appropriate solution for the same and fix it. That's how you should approach.

Reply:
I will surely try debugging it again and provide you the updates.